### PR TITLE
feat(presets): add merge confidence preset

### LINKED ISSUE #22891: Internalize Merge Confidence preset
### Describe the proposed change(s).

Deprecate the https://github.com/whitesource/merge-confidence/ repo and instead make this preset "internal".

This will require migrating it for any repos or presets representing it externally e.g. `github.com/

Propose to make it its own prefix, e.g. `mergeconfidence:default`. More can be added later, for example if we have a preset which uses more or less columns, etc.

I suggest that the column definitions can be added by default, without need for the preset - and allowing others to use them manually. Then the preset itself can just be the [packageRules part](https://github.com/whitesource/merge-confidence/blob/033f7a63112349998f04b14665068c1bca4039a5/beta.json#L8-L14)

### Describe why we need/want these change(s).

Having it hosted on GitHub leads to a lot of extra API calls

### PR DESCRIPTION
**Pull request by Gabriel-Ladzaretti**
## Changes
 - add merge confidence preset
 - update default PR definitions with badge related columns
 - copy external markdown to docs/usage

## Context

closes #22891

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by Gabriel-Ladzaretti** [comment]
related 
- https://github.com/renovatebot/renovatebot.github.io/pull/296

File: docs/usage/merge-confidence.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+### Data
+
+We plan to expose much more of the data via a companion website, such as number of users of a package and popular repositories which already updated to the version in question.
+
+## Explanations
+
+### npm package ranking
+
+npm packages less than three days old can be [unpublished](https://docs.npmjs.com/unpublishing-packages-from-the-registry/), which can result in a service impact if you have updated to a package that gets unpublished.
+This is why npm packages can only get the **High** Confidence badge when they are at least three days old.
```
Comment (rarkins) [inline docs/usage/merge-confidence.md:88]: @Gabriel-Ladzaretti this data might be outdated. I think npm silently changed it from 3 days to 7

File: docs/usage/merge-confidence.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+### Percentage values weighting
+
+The percentages for `Adoption` and `Passing` are weighted towards Organizations, private repositories, and projects with high test reliability.
+This means those values aren't _raw_ percentages.
+
+## Roadmap
+
+High-level information on planned improvements is available in the [Merge Confidence Roadmap Project](https://github.com/whitesource/merge-confidence/projects/1) on GitHub.
+
```
Comment (rarkins) [inline docs/usage/merge-confidence.md:98]:
```suggestion
```

I'd like to deprecate this as I didn't update it and it didn't get engagement

File: docs/usage/merge-confidence.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+The algorithm that decides on the values is private and is not something we plan to share.
+Similar to a search engine's algorithm, we plan to adjust and improve it over time, for example by using historical data to set a baseline confidence level for packages.
+
+### Data
+
+We plan to expose much more of the data via a companion website, such as number of users of a package and popular repositories which already updated to the version in question.
+
+## Explanations
+
+### npm package ranking
```
Comment (rarkins) [inline docs/usage/merge-confidence.md:85]:
```suggestion
### Package ranking
```

Might be a generic package ranking, not just npm?

File: docs/usage/merge-confidence.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+- npm
+- Maven
+- PyPI
+
+We plan to support more languages soon.
+
+## Enabling and disabling
+
+If you use the Mend Renovate App (formerly WhiteSource Renovate App) then the badges are enabled automatically.
```
Comment (rarkins) [inline docs/usage/merge-confidence.md:38]:
```suggestion
If you use the Mend Renovate App then the badges are enabled automatically.
```

File: docs/usage/merge-confidence.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+Merge Confidence adds the following badges to your pull requests:
+
+- **Age**: The age of the package
+- **Adoption**: The percentage of this package's users (within Renovate) which are using this release
+- **Passing**: The percentage of updates which have passing tests for this package
+- **Confidence**: The confidence level for this update
+
+## Supported platforms
+
+Merge Confidence badges for pull requests are available in beta on any supported Renovate platform, including Mend Remediate.
```
Comment (rarkins) [inline docs/usage/merge-confidence.md:24]:
```suggestion
Merge Confidence badges for pull requests are available on any supported platform or Renovate distribution, including Mend Remediate.
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/merge-confidence.md (hunk 1 above)
Comment (Gabriel-Ladzaretti) [reply docs/usage/merge-confidence.md:88]: seems it is still 3 days 🤔 
https://docs.npmjs.com/policies/unpublish#packages-published-less-than-72-hours-ago

Comment (rarkins) [reply docs/usage/merge-confidence.md:88]: Great

File: lib/config/presets/internal/merge-confidence.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+      {
+        matchDatasources: ['maven', 'npm', 'pypi'],
+        matchUpdateTypes: ['patch', 'minor', 'major'],
+        prBodyColumns: [
+          'Package',
+          'Change',
+          'Age',
+          'Adoption',
+          'Passing',
+          'Confidence',
```
Comment (rarkins) [inline lib/config/presets/internal/merge-confidence.ts:16]: I'm a little tempted to make the default Age + Confidence and then offer a second "expanded" preset which includes all four

Comment (rarkins) [reply lib/config/presets/internal/merge-confidence.ts:16]: @JamieMagee @viceice @secustor any thoughts on that?

File: docs/usage/merge-confidence.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+  ]
+```
+
+If you want to disable the badges in the app, add this to your config:
+
+```json lines
+  "ignorePresets": [
+    "mergeConfidence:default"
+  ]
+```
```
Comment (HonkingGoose) [inline docs/usage/merge-confidence.md:54]: Remove the `lines` part here as well, and run Prettier.

File: docs/usage/merge-confidence.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+We plan to support more languages soon.
+
+## Enabling and disabling
+
+If you use the Mend Renovate App then the badges are enabled automatically.
+
+If you don't use the app, you can enable the badges by adding this to your Renovate config:
+
+```json lines
```
Comment (Gabriel-Ladzaretti) [inline docs/usage/merge-confidence.md:42]: @HonkingGoose does `mkdocs` supports `json lines` ?
couldnt find anything documented about this. 

if it does not. then we should either use `json` and surround this lines with curly brackets:

```json
{
    "extends": [
        "mergeConfidence:default"
     ]
}
```
or use `js`:
```js
"extends": [
    "mergeConfidence:default"
  ]
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: docs/usage/merge-confidence.md (hunk 7 above)
Comment (Gabriel-Ladzaretti) [reply docs/usage/merge-confidence.md:54]: it fails tests
prettier fix doesnt help, we need to surround it with brackets 

```
$ node tools/check-fenced-code.mjs
Invalid JSON in fenced code block (docs\usage\merge-confidence.md lines 42-47): Unexpected token : in JSON at position 11
Invalid JSON in fenced code block (docs\usage\merge-confidence.md lines 50-55): Unexpected token : in JSON at position 17
2 issues found. Fix this manually by ensuring each block is a valid, complete JSON document. See above for lines affected.
error Command failed with exit code 1.
```

**Review by HonkingGoose — APPROVED** [review]

File: lib/config/presets/internal/merge-confidence.ts (hunk 6 above)
Comment (viceice) [reply lib/config/presets/internal/merge-confidence.ts:16]: sounds good to me

Comment (Gabriel-Ladzaretti) [reply lib/config/presets/internal/merge-confidence.ts:16]: should i split it then, or maybe in a separate PR ?

Comment (Gabriel-Ladzaretti) [reply lib/config/presets/internal/merge-confidence.ts:16]: should I split the preset then ?

Comment (rarkins) [reply lib/config/presets/internal/merge-confidence.ts:16]: Yes, let's have mergeConfidence:default and mergeConfidence:extended

Comment (rarkins) [reply lib/config/presets/internal/merge-confidence.ts:16]: Alternatively, let's keep feature parity and refer to the existing one as `full`. I'll make some suggestions

File: lib/config/presets/internal/merge-confidence.ts [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import type { Preset } from '../types';
+
+export const presets: Record<string, Preset> = {
+  default: {
```
Comment (rarkins) [inline lib/config/presets/internal/merge-confidence.ts:4]:
```suggestion
  'all-badges': {
```

File: lib/config/presets/internal/merge-confidence.ts [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+    description: 'Enable Merge Confidence badges for pull requests.',
```
Comment (rarkins) [inline lib/config/presets/internal/merge-confidence.ts:5]:
```suggestion
    description: 'Enable all Merge Confidence badges for pull requests.',
```

File: docs/usage/merge-confidence.md
Comment (rarkins) [inline docs/usage/merge-confidence.md:1]: Please replace mergeConfidence:default with mergeConfidence:all-badges

File: lib/config/presets/common.ts [hunk 11]
```diff
@@ -27,6 +27,7 @@ export const removedPresets: Record<string, string | null> = {
   'helpers:oddIsUnstable': null,
   'helpers:oddIsUnstablePackages': null,
   'group:jsTestMonMajor': 'group:jsTestNonMajor',
+  'github>whitesource/merge-confidence:beta': 'mergeConfidence:default',
```
Comment (rarkins) [inline lib/config/presets/common.ts:30]:
```suggestion
  'github>whitesource/merge-confidence:beta': 'mergeConfidence:all-badges',
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/config/presets/internal/merge-confidence.ts [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+          'Change',
+          'Age',
+          'Adoption',
+          'Passing',
+          'Confidence',
+        ],
+      },
+    ],
+  },
+  minimal: {
```
Comment (rarkins) [inline lib/config/presets/internal/merge-confidence.ts:21]:
```suggestion
  'age-confidence-badges': {
```

**Review by rarkins — DISMISSED** [review]

File: lib/config/presets/internal/merge-confidence.ts [hunk 13]
```diff
@@ -0,0 +1,31 @@
+import type { Preset } from '../types';
+
+export const presets: Record<string, Preset> = {
+  'all-badges': {
+    description: 'Enable all Merge Confidence badges for pull requests.',
```
Comment (HonkingGoose) [inline lib/config/presets/internal/merge-confidence.ts:5]:
```suggestion
    description: 'Show all Merge Confidence badges for pull requests.',
```

File: lib/config/presets/internal/merge-confidence.ts [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+          'Age',
+          'Adoption',
+          'Passing',
+          'Confidence',
+        ],
+      },
+    ],
+  },
+  'age-confidence-badges': {
+    description: 'Enable Age & Confidence badges for pull requests.',
```
Comment (HonkingGoose) [inline lib/config/presets/internal/merge-confidence.ts:22]:
```suggestion
    description: 'Show only the Age and Confidence Merge Confidence badges for pull requests.',
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
We added a new "show only Age and Confidence" preset, but the big page does not explain that.

Can you add some suitable text about the two presets?

**Review by rarkins — DISMISSED** [review]

File: docs/usage/merge-confidence.md [hunk 15]
```diff
@@ -0,0 +1,98 @@
...
+- Maven
+- PyPI
+
+We plan to support more languages soon.
+
+## Enabling and disabling
+
+If you use the Mend Renovate App then the badges are enabled automatically.
+
+If you don't use the app, you can enable the badges by adding the `mergeConfidence:all-badges` preset to the `extends` array in your Renovate config:
```
Comment (HonkingGoose) [inline docs/usage/merge-confidence.md:40]:
```suggestion
If you're self-hosting Renovate, you can enable the badges by adding the `mergeConfidence:all-badges` preset to the `extends` array in your Renovate config:
```

Let's put our points positively.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
One small change, and we're good to merge after.

**Review by rarkins — APPROVED** [review]

**Review by HonkingGoose — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**